### PR TITLE
Add github release yml file

### PR DESCRIPTION
**Pull request by cfkoehler**
This adds auto formatting to the Releases github page

### CODE REVIEW

File: .github/release.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      labels:
+        - documentation
+    - title: Automation and Testing
+      labels:
+        - CI/CD
+        - test-only
+    - title: Breaking Changes
+      labels:
+        - major
+        - integration
```
Comment (cfkoehler) [inline .github/release.yml:30]: Can remove the integration label if we think it is only relevant internally

File: .github/release.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+changelog:
```
Comment (jpdahlke) [inline .github/release.yml:1]:
```suggestion
---
changelog:
```

File: .github/release.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  categories:
+    - title: New Features
+      labels:
+        - feature
+    - title: Improvements
+      labels:
+        - enhancement
+        - framework
+        - tech-debt
+    - title: Security
```
Comment (jpdahlke) [inline .github/release.yml:11]:
```suggestion
    - title: Security
```

File: .github/release.yml (hunk 1 above)
Comment (jpdahlke) [reply .github/release.yml:30]: We can try it for now

**Review by jpdahlke — APPROVED** [review]

**Review by drivenflywheel — APPROVED** [review]

**Comment by drivenflywheel** [comment]
Reference for the curious:  https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes

**Comment by jpdahlke** [comment]
illustrative example in raft: https://github.com/NationalSecurityAgency/raft/releases/tag/0.18.0

**Pull request merged**